visualization/app/codeCharta/ui/attributeSideBar/attributeSideBarHeaderSection/nodePath/nodePath.component.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { render } from "@testing-library/angular"
+
+import { Node } from "../../../../codeCharta.model"
+import { selectedNodeSelector } from "../../../../state/selectors/selectedNode.selector"
+import { NodePathComponent } from "./nodePath.component"
+
+jest.mock("../../../../state/selectors/selectedNode.selector", () => ({
+	selectedNodeSelector: jest.fn()
+}))
+const selectedNodeSelectorMock = selectedNodeSelector as unknown as jest.Mock<ReturnType<typeof selectedNodeSelector>>
+
+describe("nodePathComponent", () => {
+	it("should display an empty p tag, if no building is selected", async () => {
+		const { container } = await render(NodePathComponent, { componentProperties: { node: undefined } })
+		const pTag = container.querySelector("p")
+
+		expect(pTag).not.toBe(null)
+		expect(pTag.textContent).toBe("")
+	})
+
+	it("should display only node path, when a file is selected", async () => {
+		const node = { isLeaf: true, path: "some/file.ts" } as unknown as Node
```

Comment:
Do we have a full actual (fixture) node that we could use instead of casting the object? The same applies to the other one.

Reply:
I see your point but I am a little torn here. I also don't like to let loose complete node type if I am only interested in two properties. I made the type of the component more specific, which is now `@Input() node?: Pick<Node, "path" | "isLeaf">`. This removes the need for the cast and I can still see inline immediately the values of the test.

Reply:
SGTM.